system/jlib/jcontainerized.cpp

```diff
@@ -224,7 +224,22 @@ bool applyYaml(const char *componentName, const char *wuid, const char *job, con
     args.append(" \"--k8sJob=true\"");
     for (const auto &p: extraParams)
     {
-        if (hasPrefix(p.first.c_str(), "_HPCC_", false)) // job yaml substitution
+        if (streq(p.first.c_str(), "_HPCC_JOB_VERSION_"))
+        {
+            const char *imagePattern = getenv("imagePattern");
+            if (imagePattern)
+            {
+                const char *imageVersionStart = strstr(imagePattern, ":");
+                if (imageVersionStart)
+                {
+                    VStringBuffer oriImagePatternSpec("image: %s", imagePattern);
+                    VStringBuffer newImagePatternSpec("image: %.*s:%s", (int)(imageVersionStart-imagePattern), imagePattern, p.second.c_str());
+                    jobYaml.replaceString(oriImagePatternSpec, newImagePatternSpec);
```

Comment:
Should this also change imagePattern: xxxxxx to simplify the logic in thor?

Reply:
I am intentionally leaving imagePattern as the untouched original version, so that if 1st job comes in is with manipulated image: (based on platformVersion), and then subsequent jobs come in without a platformVersion directive, the instance can detect the mismatch, i.e. that the thor instance is based on the 1st job's platformVersion, and the 2nd job doesn't specify one, so is assumed to want the "standard" base image version (original helm generated image version - which is still within imagePattern and env var.)